### ISSUE TITLE
It seems like class member initializations sometimes aren't in the right block of code.

### ISSUE THREAD

**Issue by leaf-node** [Cleod9/as3js#14]
The following code

````
package Box2D.Collision
{
import Box2D.Collision.*;
import Box2D.Collision.Shapes.*;
import Box2D.Common.*;
import Box2D.Common.Math.*;

internal class b2Simplex
{

public function b2Simplex()
{
        m_vertices[0] = m_v1;
        m_vertices[1] = m_v2;
        m_vertices[2] = m_v3;
}

...

public var m_v1:b2SimplexVertex = new b2SimplexVertex();
public var m_v2:b2SimplexVertex = new b2SimplexVertex();
public var m_v3:b2SimplexVertex = new b2SimplexVertex();
public var m_vertices:Array/*b2SimplexVertex*/ = new Array/*b2SimplexVertex*/(3);
public var m_count:int;
}

}
````

... is transpiled into:

````
    };Program["Box2D.Collision.b2Simplex"]=function(module, exports){  var b2SimplexVertex, b2Math, b2Vec2, b2Settings;
    module.inject = function () {
      b2SimplexVertex = module.import('Box2D.Collision', 'b2SimplexVertex');
      b2Math = module.import('Box2D.Common.Math', 'b2Math');
      b2Vec2 = module.import('Box2D.Common.Math', 'b2Vec2');
      b2Settings = module.import('Box2D.Common', 'b2Settings');
    };

    var b2Simplex = function() {
          this.m_v1 = new b2SimplexVertex();
          this.m_v2 = new b2SimplexVertex();
          this.m_v3 = new b2SimplexVertex();
          this.m_vertices = new Array/*b2SimplexVertex*/(3);
      this.m_vertices[0] = this.m_v1;
      this.m_vertices[1] = this.m_v2;
      this.m_vertices[2] = this.m_v3;
    };
````

... which causes a `TypeError: b2SimplexVertex is not a constructor` error. I compared this with similar case that seems to work fine:

````
package Box2D.Collision {

import Box2D.Common.b2ClassIDs;

public class b2Bound{

        private static var class_ids = new b2ClassIDs();
...
        static public var class_id = b2Bound.class_ids.e_class_id_b2Bound;
}
}
````

````
    };Program["Box2D.Collision.b2Bound"]=function(module, exports){  var b2ClassIDs;
    module.inject = function () {
      b2ClassIDs = module.import('Box2D.Common', 'b2ClassIDs');
      b2Bound.class_ids = new b2ClassIDs();
      b2Bound.class_id = b2Bound.class_ids.e_class_id_b2Bound;
    };
````

For context: https://gitlab.com/sudoman/box2dflash2js/tree/890a6f1e4148636937fbdf4fa323bb040030ee85

**Comment by Cleod9** [comment]
This one's a bit trickier than the others, will let you know what i find

**Comment by Cleod9** [comment]
After some research I believe I've uncovered the issue with this one. Looks like `inject()` alone may not be sufficient for the initialization process. In AS3/AVM2 there are these hidden methods called `$cinit` and `$init` that are called before an instance of a class is first used (but seemingly after the linking occurs). What I'll probably have to do is place another loop to handler static initialization `$cinit` after the injection process occurs. It so happens that `$init` is more or less already covered since I place those initializers at the top of the constructor. (Though perhaps i may move those into a private `$init` to make it more consistent with AS3)

Will update once I get around to making the updates

**Comment by Cleod9** [comment]
@sudoman 

This project is essentially dead so I'm going to close this, but I did one last update `0.3.3` which does partially address your issue. I was actually finally able to get your project to build, though there were a few places I had to manually insert the class name in front of some static members (e.g. `MyClass.staticMember`). AS3JS is not intelligent enough to infer those in all situations

Take care!
